src/SME.SGP.Aplicacao/Commands/PlanejamentoAnual/Copiar/MigrarPlanejamentoAnualCommandHandler.cs

```diff
@@ -0,0 +1,77 @@
+        public MigrarPlanejamentoAnualCommandHandler(IMediator mediator,
+                                              IUnitOfWork unitOfWork) : base(mediator)
+        {
+            this.unitOfWork = unitOfWork ?? throw new System.ArgumentNullException(nameof(unitOfWork));
+        }
+
+        public async Task<bool> Handle(MigrarPlanejamentoAnualCommand comando, CancellationToken cancellationToken)
+        {
+            unitOfWork.IniciarTransacao();
+            var periodosOrigem = await mediator.Send(new ObterPlanejamentoAnualPeriodosEscolaresCompletoPorIdQuery(comando.Planejamento.PlanejamentoPeriodosEscolaresIds.ToArray()));
+            var usuario = await mediator.Send(new ObterUsuarioLogadoQuery());
+
+            List<string> excessoes = new List<string>();
+
+            if (!periodosOrigem.Any())
+                throw new NegocioException($"Nenhum período foi encontrado");
+
+            // Validando as turmas
+            foreach (var turma in comando.Planejamento.TurmasDestinoIds)
+            {
+                var checarTurma = await mediator.Send(new ObterTurmaComUeEDrePorIdQuery(turma));
+                if (checarTurma == null)
+                    throw new NegocioException($"Turma não encontrada");
+
+                foreach (var periodoOrigem in periodosOrigem)
+                {
+                    var periodo = await mediator.Send(new ObterPeriodoEscolarePorIdQuery(periodoOrigem.PeriodoEscolarId));
+                    var temAtribuicao = await mediator.Send(new ObterUsuarioPossuiPermissaoNaTurmaEDisciplinaNoPeriodoQuery(comando.Planejamento.ComponenteCurricularId, checarTurma.CodigoTurma, usuario.CodigoRf, periodo.PeriodoInicio.Date, periodo.PeriodoFim.Date));
+                    if (!temAtribuicao)
+                        excessoes.Add($"Você não possui atribuição na turma {checarTurma.Nome} - {periodo.Bimestre}° Bimestre.");
+
+                    var periodoEmAberto = mediator.Send(new TurmaEmPeriodoAbertoQuery(checarTurma, DateTime.Today, periodo.Bimestre, checarTurma.AnoLetivo == DateTime.Today.Year)).Result;
+                    if (!periodoEmAberto)
+                        excessoes.Add($"O {periodo.Bimestre}° Bimestre não está aberto.");
+                }
+
+                var planejamentoCopiado = new PlanejamentoAnual(checarTurma.Id, comando.Planejamento.ComponenteCurricularId);
+                planejamentoCopiado.PeriodosEscolares.AddRange(periodosOrigem);
+
+                if (!excessoes.Any())
+                    await mediator.Send(new SalvarCopiaPlanejamentoAnualCommand(planejamentoCopiado));
+            }
+
+            if (excessoes.Any())
+            {
+                var str = new StringBuilder();
+                str.AppendLine($"Os seguintes erros foram encontrados: ");
+                foreach (var t in excessoes)
+                {
+                    str.AppendLine($"- {t}");
+                }
+                unitOfWork.Rollback();
```

Comment:
tem que por o rollback dentro de um catch pra controle de exceção.. no código ai tem um throw se a turma não for encontrada..

Caso tu consiga gravar a copia do planejamento de uma turma e na próxima discparar exceção.. o rollaback não será realizado